magic.js

```diff
@@ -1260,6 +1260,53 @@ module.exports.util.timingSafeCompare = (input, ref) => {
   return cnstcomp(inputBuffer, refBuffer) && inputLength === refLength;
 };
 
+
+/*
+ * rsaKeypairGen
+ *
+ * Get an RSA private/public keypair
+ *
+ * @function
+ * @api public
+ *
+ * @param {Function} cb
+ * @returns {Callback|Promise}
+ */
+
+module.exports.util.rsaKeypairGen = (cb) => {
+  const done  = ret(cb)
+
+  if (crypto.generateKeyPair) { // node >= 10
+    return new Promise((resolve, reject) => {
+      crypto.generateKeyPair('rsa', {
+        modulusLength: 2048,
+        publicExponent: 65537,
+        publicKeyEncoding: {
+          type: 'spki',
```

Comment:
@esarafianou this is probably the reason why you're seeing format discrepancy.  If you want the

>  -----BEGIN RSA PUBLIC KEY-----

 format, try changing type to pkcs1